packages/lib/contracts/upgradeability/Proxy.sol

```diff
@@ -28,23 +28,10 @@ contract Proxy {
    * @param implementation Address to delegate.
    */
   function _delegate(address implementation) internal {
+    (bool success, bytes memory data) = implementation.delegatecall(msg.data);
+    require(success, string(data));
```

Comment:
This is slightly more complicated because Solidity will add a prefix to the reason in this `require`, but `data` already contains this prefix. The way to work around this is using `revert` in assembly, which should look the same as the `return` statement in the assembly block below.

Reply:
Can we use ‘abi.decode(data, (string))`?

Reply:
No that wouldn't work. Solidity encodes the data as if it were a function call, with a 4-byte selector at the beginning.

Reply:
@frangio wait, I expected `data` to contain revert reason. Can we do parse the prefix like this `abi.decode(data, (bytes4,string))`?

Reply:
IIRC That doesn't work because it tries to decode it as if it were function arguments, which are padded differently. It's possible to use the new slices feature from 0.6 but these contracts are still on 0.5. See https://solidity.readthedocs.io/en/latest/060-breaking-changes.html#new-features.

Reply:
Yep, found it: https://ethereum.stackexchange.com/a/66404/3032